src/applications/personalization/profile-2/components/ProfileWrapper.jsx

```diff
@@ -1,28 +1,93 @@
-import React from 'react';
+import React, { useEffect } from 'react';
 import { connect } from 'react-redux';
+import LoadingIndicator from '@department-of-veterans-affairs/formation-react/LoadingIndicator';
 
 import RequiredLoginView from 'platform/user/authorization/components/RequiredLoginView';
 import backendServices from 'platform/user/profile/constants/backendServices';
 
+import {
+  fetchMilitaryInformation as fetchMilitaryInformationAction,
+  fetchHero as fetchHeroAction,
+} from 'applications/personalization/profile360/actions';
+
 import ProfileHeader from './ProfileHeader';
 import ProfileSideNav from './ProfileSideNav';
 
-const ProfileWrapper = ({ children, user }) => (
-  <RequiredLoginView serviceRequired={backendServices.USER_PROFILE} user={user}>
-    <ProfileHeader />
-    <div className="usa-grid usa-grid-full">
-      <div className="usa-width-one-fourth">
-        <ProfileSideNav />
-      </div>
-      <div className="usa-width-three-fourths">{children}</div>
+const ProfileWrapper = ({
+  children,
+  fetchFullName,
+  fetchMilitaryInformation,
+  showLoader,
+  user,
+}) => {
+  useEffect(
+    () => {
+      fetchMilitaryInformation();
+      fetchFullName();
+    },
+    [fetchMilitaryInformation, fetchFullName],
+  );
+
+  // content to show if the component is waiting for data to load
+  const loadingContent = (
```

Comment:
This loading content matches the content shown by the `RequiredLoginView` so  when `RequiredLoginView` stops showing its spinner this spinner will seamlessly take its place.